### ISSUE TITLE
Remove everything related with a page object form cache

### ISSUE THREAD

**Issue by leotiger** [getgrav/grav#2093]
I'm working on load optimized website which includes shop functionality including stock management. In this context I update on checkout stocks and this naturally has an impact on the page but also on the parent which serves as a product index. 

In the page class we dispose of the ``$page->cachePageContent()`` function which allows to update the content part and meta data but this is not enough to force a refill of the cache when the page is requested again.

What I can do, this works, is to use the static function clearCache inside of the Cache class but this wipes out all cached pages. Hints? Am I missing something?

**Comment by leotiger** [comment]
Update: Using clearCache with cache-only, like I did, works with APCu as caching service but not with Memcached and probably not with redis. With memcached only the default clearCache or with all achieves that the page refreshes.

**Comment by leotiger** [comment]
More background: cache-only works only for file cache. APCu was not active due to be volatile with cli option on for cache in the configuration. This means, a partial cache delete only works with expected results when cache uses file mode, at least that's the result of my tests. Probably I don't have enough insight into caching mechanisms but as this uses doctrine as an abstraction one would expect that at least the exposed functionality of clearing the cache works for the options 'all', 'cache-only', 'tmp-only', etc. the same and that's not the case.

**Comment by leotiger** [comment]
And what works with apcu is a two-liner:

```
$user_config = USER_DIR . 'config/system.yaml';
touch($user_config);
```

No direct calls of clear cache. With cache_method set to none and by calling this two liner in code when touching header or other stuff of a page, I obtain a correct page refresh. Compared to clearCache('all') it's fast. For me good enough.

**Comment by leotiger** [comment]
I close this. Nevertheless: would be great if somebody explained to me if there exist approaches which allow a finer granularity. But from what I understand a great deal of pages and routes information is cached as a massive object...

**Comment by mahagr** [comment]
Yes, Grav is set to clear any cache based on the timestamp of `config/system.yaml`. This is because of with APCu (and Memcached etc) Grav currently uses separate cache in CLI/site/admin, though clearing the cache from any of those locations do not affect caching in any other locations. I don't know why it is done in that way as it doesn't make any sense to me to do that -- that said, you do want to separately cache twig and admin related stuff...

I am currently working towards better caching for Grav 2.0, but it is still under development and some of the concepts haven't been implemented, either. But we have tested the new way in few sites up to thousand pages and it greatly improves performance as it allows much more granual caching and has better change detection mechanism, too.

**Comment by leotiger** [comment]
@mahagr Thx for the feedback. Took me several hours investigating the whole stuff. There are some bits of the related problems exposed in old issues but I think a good intermediate solution would be to add the two liner as an option to the clearCache function allowing to simply touch the timestamp and to execute the touch based on the selected / effective caching driver. And to document this, while 2.0 is not out.

**Comment by leotiger** [comment]
To put it clear: the big advantage, if the configuration is ok, there's no need to delete and recreate files on disk this way. The profile for "touch-only" can bypass all disk operations but provokes a memcached and apcu to refresh.

**Comment by leotiger** [comment]
Even better, if a user has cache_method set to "none" and with an additional option this could be triggered automatically on save of a page object which removes the need to clear the cache manually.

**Comment by leotiger** [comment]
@mahagr, @rhukster Reopen if you feel that this is worth thinking about. The "none" setting for the detection of changes is great for spead, but it demands additional steps and obliges not to forget to clear the cache manually. With an automated "touch-only" method, to be activated by a configuration setting, this in conjunction with memcached or apcu makes life a lot easier.

**Comment by mahagr** [comment]
@rhukster can decide what to do with this issue. :)

**Comment by rhukster** [comment]
Thanks for this @leotiger, I'll review the PR when I get a mo...  Bit swamped the next couple of days.
